### PR TITLE
Simplify examples which use `ContractRef`

### PR DESCRIPTION
**Pull request by HCastano**
The `Delegator`, now `MultiContractRef`, example has always been a bit too complicated to
really highlight the basics of how to use `ContractRef`s for cross-contract calls.

This PR introduces a set of contracts which shows the bare minimum needed for performning
cross-contract calls using `ContractRef`s. 

It also renames the `Delegator` contract to `MultiContractRef`. This was done because it
the term "delegate" should be reserved for contracts which use delegate calls. This has
also caused some confusion recently.

### CODE REVIEW

File: integration-tests/multi_contract_ref/README.md [hunk 1]
```diff
@@ -1,13 +1,14 @@
-# Delegator Smart Contract
+# `MultiContractRef` Smart Contract
```
Comment (SkymanOne) [inline integration-tests/multi_contract_ref/README.md:1]: The name can be confused with the one automatically generated by `ContractRef`.
I would name it smth like 
```suggestion
# `MultiContractCaller` Smart Contract
```
or just
```suggestion
# `MultiContract` Smart Contract
```

File: integration-tests/basic_contract_ref/lib.rs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+#[ink::contract]
+mod basic_contract_ref {
+    /// We import the generated `ContractRef` of our other contract.
+    ///
+    /// Note that the other contract must have re-exported it (`pub use OtherContractRef`) for us
+    /// to have access to it.
+    use other_contract::OtherContractRef;
+
+    #[ink(storage)]
+    pub struct BasicContractRef {
```
Comment (SkymanOne) [inline integration-tests/basic_contract_ref/lib.rs:12]: Similarly here `BasicContractRef` is confusing. 
```suggestion
    pub struct BasicContract {
```

**Review by SkymanOne — COMMENTED** [review]
Looks good, although I don't think naming a contract `<ContractName>Ref` is a good idea as it can be confused with automatically generated `ContractRef` reference.
We should discourage devs from using that naming in their projects.

**Review by SkymanOne — APPROVED** [review]
Thanks!

**Review by sergejparity — APPROVED** [review]

**Comment by cmichi** [comment]
@HCastano Please transfer to our `ink-examples` repo as well.

**Pull request merged**